src/main.c

```diff
@@ -58,6 +61,21 @@ struct euicc_ctx euicc_ctx = {0};
 
 void main_init_euicc()
 {
+    const char *custom_aid_str = getenv("LPAC_CUSTOM_ISD_R_AID");
+    if (custom_aid_str)
+    {
+        unsigned char custom_aid[ISD_R_AID_STR_LENGTH];
+        const int custom_aid_len = euicc_hexutil_hex2bin(custom_aid, ISD_R_AID_STR_LENGTH, custom_aid_str);
+        if (custom_aid_len != ISD_R_AID_STR_LENGTH)
+        {
+            jprint_error("euicc_init", "invalid custom AID given");
+            exit(-1);
+        }
```

Comment:
If the user gives a hex string larger than 16B, it will be truncated silently. This is not a good practice. Please check the length before passing the hex string into `euicc_hexutil_hex2bin()`.

Reply:
Isn't this actually checked in `euicc_hexutil_hex2bin` already?

Reply:
Sorry, I read the wrong line.